pkg/pillar/cas/containerd.go

```diff
@@ -167,6 +159,23 @@ func (c *containerdCAS) IngestBlob(blobs ...*types.BlobStatus) ([]*types.BlobSta
 			sha = fmt.Sprintf("%s:%s", digest.SHA256, strings.ToLower(blob.Sha256))
 		)
 
+		log.Infof("IngestBlob(%s): processing blob %+v", blob.Sha256, blob)
+		// Process the blob only if its not in a loaded status already
+		if blob.State == types.LOADED {
+			log.Infof("IngestBlob(%s): Not loading blob as it is already marked as loaded", blob.Sha256)
+			loadedBlobs = append(loadedBlobs, blob)
+			continue
+		}
+
+		// Process the blob only if it isn't already loaded. It might be loaded but not
+		// yet marked in the blobstatus
+		if info, err := containerd.CtrGetBlobInfo(sha); err != nil && info.Digest.String() == sha {
+			log.Infof("IngestBlob(%s): Not loading blob as it is already loaded in storage", blob.Sha256)
+			loadedBlobs = append(loadedBlobs, blob)
```

Comment:
This seems wrong - should wait for it to be marked as LOADED.

Reply:
Going back to the previous comment. How do we do the wait?